src/truncated/lognormal.jl

```diff
@@ -0,0 +1,52 @@
+# Moments of the truncated log-normal can be computed directly from the moment generating
+# function of the truncated normal:
+#   Let Y ~ LogNormal(μ, σ) truncated to (a, b). Then log(Y) ~ Normal(μ, σ) truncated
+#   to (log(a), log(b)), and E[Y^n] = E[(e^log(Y))^n] = E[e^(nlog(Y))] = mgf(log(Y), n).
+
+# Given `truncate(LogNormal(μ, σ), a, b)`, return `truncate(Normal(μ, σ), log(a), log(b))`
+function _truncnorm(d::Truncated{<:LogNormal})
+    μ, σ = params(d.untruncated)
+    T = partype(d)
+    a = d.lower === nothing ? nothing : log(T(minimum(d)))
+    b = d.upper === nothing ? nothing : log(T(maximum(d)))
```

Comment:
Is there a reason to use `minimum`/`maximum` instead of `d.lower`/`d.upper`?

Reply:
I feel like maybe I had a reason but who knows what it was

Reply:
Ah, I remember now: it was to handle truncation points outside of the support, e.g. `truncated(LogNormal(); lower=-1)`. In that case, `d.lower == -1` but `minimum(d) == 0`. I should add a test for that.